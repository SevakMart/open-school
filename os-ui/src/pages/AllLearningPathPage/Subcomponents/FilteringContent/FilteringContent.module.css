.filteringContent {
  margin-top: 18px;
}

.filteringSubContent {
  margin-top: 16px;
}

.mainFilteringContentTitle {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  color: #333941;
}

.filteringSubContentTitle {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  color: #333941;
}/*# sourceMappingURL=FilteringContent.module.css.map */