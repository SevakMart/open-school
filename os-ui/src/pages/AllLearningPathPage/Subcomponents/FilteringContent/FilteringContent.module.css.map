{"version":3,"sources":["FilteringContent.module.scss","FilteringContent.module.css","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss","../../../../styles/Configs/Colors.scss"],"names":[],"mappings":"AAKA;EACI,gBAAA;ACJJ;;ADMA;EACI,gBAAA;ACHJ;;ADKA;EEVE,yBCDY;EDEZ,kBFUmC;EETnC,gBFS0C;EER1C,eFQ8C;EEP9C,iBFOmD;EACjD,cIRe;AHUnB;;ADAA;EEdE,yBCDY;EDEZ,kBFcmC;EEbnC,gBFa0C;EEZ1C,eFY8C;EEX9C,iBFWmD;EACjD,cIZe;AHmBnB","file":"FilteringContent.module.css"}