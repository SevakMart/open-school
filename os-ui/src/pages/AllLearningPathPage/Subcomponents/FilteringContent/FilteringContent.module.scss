@import "../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../styles/Mixins/FontMixin.scss";

.filteringContent .filteringSubContent{
    @include flexStyle(column,flex-start,space-evenly);
}
.mainFilteringContentTitle{
    @include fontStyle('Montserrat',normal,500,1rem,1.5rem);
    color:#333941;
}
.filteringSubContentTitle{
    @include fontStyle('Montserrat',normal,500,0.75rem,1rem);
    color:#333941;
}