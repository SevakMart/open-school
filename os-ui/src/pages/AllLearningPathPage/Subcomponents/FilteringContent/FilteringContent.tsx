import styles from './FilteringContent.module.scss';
import CheckedContent from '../CheckedContent/CheckedContent';
import { Subtype } from '../../../../types/FilteringFeaturesType';

/* eslint-disable max-len */

const FilteringContent = ({ title, content, filterFeature }:
  {title:string, content:Subtype, filterFeature:string}) => {
  const {
    filteringContent, filteringSubContent,
    mainFilteringContentTitle, filteringSubContentTitle,
  } = styles;

  return (
    <div className={filteringContent}>
      <p data-testid={title} className={mainFilteringContentTitle}>{title}</p>
      {content && Object.entries(content).map((innerContent, index) => {
        if (typeof innerContent[1] === 'object') {
          return (
            <div className={filteringSubContent} key={index}>
              <p data-testid={innerContent[0]} className={filteringSubContentTitle}>{innerContent[0]}</p>
              {Object.entries(innerContent[1]).length > 0
                && Object.entries(innerContent[1]).map((subcontent) => (
                  <CheckedContent
                    key={subcontent[1]}
                    filterFeature={filterFeature}
                    id={subcontent[0]}
                    checkedContent={subcontent[1]}
                  />
                ))}
            </div>
          );
        }
        return (
          <CheckedContent
            key={index}
            filterFeature={filterFeature}
            id={innerContent[0]}
            checkedContent={innerContent[1]}
          />
        );
      })}
    </div>
  );
};
export default FilteringContent;
