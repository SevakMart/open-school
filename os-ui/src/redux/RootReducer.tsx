import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
