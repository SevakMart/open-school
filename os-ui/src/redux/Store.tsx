import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
