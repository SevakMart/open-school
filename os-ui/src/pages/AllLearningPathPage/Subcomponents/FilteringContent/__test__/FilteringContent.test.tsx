import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../../redux/Store';
import FilteringContent from '../FilteringContent';

const mockUseNavigate = jest.fn();
const mockUseLocation = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
  useLocation: () => mockUseLocation,
}));

const contentAsString = {
  1: 'French',
  2: 'English',
  3: 'Arabic',
};
const contentAsObject = {
  JS: {
    1: 'React',
    2: 'Angular',
    3: 'Vue Js',
  },
  Android: {
    1: 'Java',
    2: 'Kotlin',
  },
};

describe('Create test cases for filteringcontent component where the content is of type string', () => {
  test('Create a snapshot case', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <FilteringContent
          title="Languages"
          content={contentAsString}
          filterFeature="languageIds"
        />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
describe('Create test cases where the content element is of type object', () => {
  test('Create a snapshot case', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <FilteringContent
          title="Category"
          content={contentAsObject}
          filterFeature="subCategoryIds"
        />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
