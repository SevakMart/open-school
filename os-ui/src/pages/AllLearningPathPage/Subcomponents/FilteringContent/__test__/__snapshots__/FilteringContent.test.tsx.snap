// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for filteringcontent component where the content is of type string Create a snapshot case 1`] = `
<DocumentFragment>
  <div
    class="filteringContent"
  >
    <p
      class="mainFilteringContentTitle"
      data-testid="Languages"
    >
      Languages
    </p>
    <div
      class="checkedContentClass"
    >
      <img
        alt="logo"
        src="CategoriesUnchecked.svg"
      />
      <p>
        French
      </p>
    </div>
    <div
      class="checkedContentClass"
    >
      <img
        alt="logo"
        src="CategoriesUnchecked.svg"
      />
      <p>
        English
      </p>
    </div>
    <div
      class="checkedContentClass"
    >
      <img
        alt="logo"
        src="CategoriesUnchecked.svg"
      />
      <p>
        Arabic
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Create test cases where the content element is of type object Create a snapshot case 1`] = `
<DocumentFragment>
  <div
    class="filteringContent"
  >
    <p
      class="mainFilteringContentTitle"
      data-testid="Category"
    >
      Category
    </p>
    <div
      class="filteringSubContent"
    >
      <p
        class="filteringSubContentTitle"
        data-testid="JS"
      >
        JS
      </p>
      <div
        class="checkedContentClass"
      >
        <img
          alt="logo"
          src="CategoriesUnchecked.svg"
        />
        <p>
          React
        </p>
      </div>
      <div
        class="checkedContentClass"
      >
        <img
          alt="logo"
          src="CategoriesUnchecked.svg"
        />
        <p>
          Angular
        </p>
      </div>
      <div
        class="checkedContentClass"
      >
        <img
          alt="logo"
          src="CategoriesUnchecked.svg"
        />
        <p>
          Vue Js
        </p>
      </div>
    </div>
    <div
      class="filteringSubContent"
    >
      <p
        class="filteringSubContentTitle"
        data-testid="Android"
      >
        Android
      </p>
      <div
        class="checkedContentClass"
      >
        <img
          alt="logo"
          src="CategoriesUnchecked.svg"
        />
        <p>
          Java
        </p>
      </div>
      <div
        class="checkedContentClass"
      >
        <img
          alt="logo"
          src="CategoriesUnchecked.svg"
        />
        <p>
          Kotlin
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
